slimta/envelope.py

```diff
@@ -113,31 +128,36 @@ def flatten(self):
         :returns: Tuple of two strings: ``(header_data, message_data)``
 
         """
-        outfp = cStringIO.StringIO()
+        outfp = cStringIO()
         Generator(outfp).flatten(self.headers, False)
         header_data = re.sub(_LINE_BREAK, '\r\n', outfp.getvalue())
         return header_data, self.message
 
     def _encode_parts(self, header_data, msg_data, encoder):
-        """Encodes any MIME part in the current message that is 8-bit."""
+        """Encodes any MIME part in the current message that is 8-bit.
+
+        :type header_data: :py:obj:`bytes`
+        :type msg_data: :py:obj:`bytes`
+        """
```

Comment:
Yeah, for better or for worse my general philosophy is to have awesome sphinx-style usage documentation on all public-face classes, methods, etc, but I worry that such usage information will encourage the unsupported use of private methods and ultimately make it harder to upgrade.